mtp_api/apps/notification/management/commands/send_notification_report.py

```diff
@@ -0,0 +1,297 @@
+import csv
+import datetime
+import pathlib
+import shutil
+import tempfile
+import zipfile
+
+from anymail.message import AnymailMessage
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from django.core.management import BaseCommand, CommandError
+from django.core.validators import validate_email
+from django.utils import timezone
+from django.utils.dateparse import parse_date
+from mtp_common.tasks import default_from_address
+
+from credit.constants import CREDIT_STATUS
+from credit.models import Credit, LOG_ACTIONS as CREDIT_LOG_ACTIONS
+from disbursement.constants import DISBURSEMENT_METHOD, DISBURSEMENT_RESOLUTION
+from disbursement.models import Disbursement, LOG_ACTIONS as DISBURSEMENT_LOG_ACTIONS
+from notification.rules import RULES, MonitoredRule
+from transaction.utils import format_amount
+
+
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        super().add_arguments(parser)
+        parser.add_argument('--since', help='Since date (inclusive)')
+        parser.add_argument('--until', help='Until date (exclusive)')
+        parser.add_argument('--rules', nargs='*', choices=RULES.keys(), help='Notification rule codes')
+        parser.add_argument('emails', nargs='+', help='Email addresses to send reports to')
+
+    def handle(self, **options):
+        emails = options['emails']
+        for email in emails:
+            try:
+                validate_email(email)
+            except ValidationError:
+                self.stderr.write(f'"{email}" is not valid email address')
```

Comment:
Could this be replaced by raising a `CommandError` like in the other cases maybe?